### PR TITLE
Alternative XQuery processor

### PR DESCRIPTION
**Pull request by adam-vessey**

### CODE REVIEW

File: includes/basex.inc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    $id = 'basex-executable-wrapper';
+
+    return array(
+      'executable' => array(
+        '#prefix' => "<div id='$id'>",
+        '#suffix' => '</div>',
+        '#type' => 'textfield',
+        '#title' => t('BaseX exectuable'),
+        '#description' => t('BaseX will be used to execute XQuery queries.<br/>!msg', array(
+          '!msg' => islandora_executable_available_message($executable))
```
Comment (willtp87) [inline includes/basex.inc]: 110 | ERROR | [x] Closing parenthesis of array declaration must be
     |       |     on a new line

File: includes/basex.inc [hunk 2]
```diff
@@ -0,0 +1,130 @@
+<?php
+
+class IslandoraXqueryBaseX implements IslandoraXqueryConfigurableProcessorInterface {
```
Comment (willtp87) [inline includes/basex.inc:6]: 3 | ERROR | [x] Missing class doc comment

File: includes/processor_admin.form.inc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  );
+
+  $form['processor'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Processor config'),
+  );
+
+  foreach ($form_state['storage']['implementations'] as $name => $class) {
+    $form['processor'][$name] = (is_callable(array($class, "getForm")) ?
+      call_user_func_array(array($class, "getForm"), array(array('processor', $name), &$form_state)) :
```
Comment (willtp87) [inline includes/processor_admin.form.inc]: 40 | ERROR | If the line declaring an array spans longer than 80
    |       | characters, each element should be broken into its own
    |       | line

File: includes/IslandoraXqueryNoExectuableException.inc [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  /**
+   * Getter for contextual info.
+   */
+  public function getImplementationInfo() {
+    return array(
+      $this->name,
+      $this->class,
+    );
+  }
```
Comment (willtp87) [inline includes/IslandoraXqueryNoExectuableException.inc:27]: 27 | ERROR | [x] Expected 1 blank line after function; 0 found

File: includes/xquery.inc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+   * List the implementations.
+   *
+   * @return array
+   *   A mapping of machine names to class names implementing
+   *   IslandoraXQueryQueryInterface.
+   */
+  public static function getImplementations() {
+    module_load_include('inc', 'islandora_xquery', 'includes/zorba');
+    return array(
+      'zorba' => \Islandora\Zorba\Options::class,
```
Comment (willtp87) [inline includes/xquery.inc:36]: 33 | ERROR | [x] Namespaced classes/interfaces/traits should be
    |       |     referenced with use statements

File: includes/basex.inc (hunk 1 above)
Comment (adam-vessey) [reply includes/basex.inc]: [Is copypasta](https://github.com/discoverygarden/islandora_xquery/blob/2deb7f281c1f7afb75ffcc74a5e339b4bc8d7e55/includes/admin.form.inc#L47)... shrug?

**Pull request merged**